hal/spi_api.h

```diff
@@ -84,7 +84,7 @@ typedef struct {
     bool        support_slave_mode; /**< If true, the device can handle SPI slave mode using hardware management on the specified ssel pin. */
     bool        hw_cs_handle; /**< If true, in SPI master mode Chip Select can be handled by hardware. */
     bool        async_mode; /**< If true, in async mode is supported. */
-
+    bool        tx_rx_buffers_equal_length; /**< If true, rx and tx buffers must have the same length. */
```

Comment:
Why is this needed? isn't this always supported ? How this relates to the hw capabilities?

Reply:
Maybe the name is misleading. Should be the oposit tx_rx_diffrent_lenght. The problem is that on some platforms we can execute spi transfer like:
`spi_master_block_write(&spi, p_tx, 10, p_rx, 5, 0xF5);`
`spi_master_block_write(&spi, p_tx, 5, p_rx, 10, 0xF5);`
Unfortunatelly on some platform this won't work. The example is `K64F` - will take max from rx/tx lenghts for transfer.

On the feature branch, we tried to handle this and this was quite tricky or even impossible (in case of async mode), so we decided to add a requirement that in async mode rx/tx length must be equal.

Here, I didn't have time to update the drivers, so I added this capability. This way I can test different sizes only on targets that support this. From my point of view allowing different tx/rx lengths is redundant, but we can't simply change the API.